dbusmock/testcase.py

```diff
@@ -63,9 +63,9 @@ def get_services_dir(cls, system_bus: bool = False) -> str:
 
     @classmethod
     def tearDownClass(cls):
-        if cls._DBusTestCase__datadir:
-            shutil.rmtree(cls._DBusTestCase__datadir)
-            cls._DBusTestCase__datadir = None
+        if DBusTestCase._DBusTestCase__datadir:
```

Comment:
Oh yes, well spotted -- that really wants to be a global variable, back then I wasn't aware that `cls` would actually be a different object for a derived class.